corehq/apps/app_manager/suite_xml.py

```diff
@@ -920,8 +926,63 @@ def details(self):
 
                         if detail_column_infos:
                             if detail.custom_xml:
-                                d = load_xmlobject_from_string(detail.custom_xml, xmlclass=Detail)
+                                d = load_xmlobject_from_string(
+                                    detail.custom_xml,
+                                    xmlclass=Detail
+                                )
                                 r.append(d)
+
+                            elif detail.use_case_tiles:
+
+                                template_args = {
+                                    "detail_id": self.id_strings.detail(module, detail_type),
+                                    "title_text_id": self.id_strings.detail_title_locale(
+                                        module, detail_type
+                                    )
+                                }
+
+                                # Get field/case property mappings
+                                for template_field in ["header", "top_left", "sex", "bottom_left", "date"]:
+                                    column = None
+                                    for col in detail.columns:
+                                        if col.case_tile_field == template_field:
+                                            column = col
+                                            break
+
+                                    template_args[template_field] = {
+                                        "prop_name": get_column_xpath_generator(
+                                            self.app, module, detail, column
+                                        ).xpath,
+                                        "locale_id": self.id_strings.detail_column_header_locale(
+                                            module, detail_type, column,
+                                        ),
+                                        # Just using default language for now
+                                        # The right thing to do would be to reference the app_strings.txt I think
+                                        "prefix": escape(
+                                            column.header.get(self.app.default_language, "")
+                                        )
+                                    }
+                                    if column.format == "enum":
```

Comment:
Looks like `column` could be `None` here. What do you think about writing some tests to exercise this code?